Refactor syntax unicode mblmedia

livecodefraser opened the pull request.

Commit:
Change MCSystemPlaySound to take a StringRef

Both iOS and Android platforms changed.

Commit:
Change MCSystemPlayVideo to take a StringRef

Both Android and iOS platforms have been modified.

Commit:
Updated MCSystemGetPlayingSound to return a StringRef

The s_sound_file global variable is now assigned an empty string when no sound is playing rather than being nil.

Commit:
Restore accidentally removed behaviour in MCMultimediaEvalSound

The pre-refactor behaviour was to return the string "done" when no sound is playing; this was accidentally removed.

runrevmark commented on engine/src/cmdse.cpp:
```diff
@@ -2116,11 +2116,13 @@ Exec_stat MCMM::exec(MCExecPoint &ep)
 	if (video)
```
The code inside LEGACY_EXEC ifdefs should be left as it was - it's used to compare for changes with the old engine branch.

runrevmark commented on engine/src/exec-multimedia.cpp:
```diff
@@ -207,13 +207,15 @@ void MCMultimediaEvalMCISendString(MCExecContext& ctxt, MCStringRef p_command, M
-	extern bool MCSystemGetPlayingSound(const char *& r_sound);
-	const char *t_sound;
-	if (MCSystemGetPlayingSound(t_sound))
+	extern void MCSystemGetPlayingSound(MCStringRef r_sound);
```
I think this extern should be MCStringRef&

runrevmark commented on engine/src/funcs.cpp:
```diff
@@ -7665,7 +7665,7 @@ Exec_stat MCSound::eval(MCExecPoint &ep)
 {
 #ifdef /* MCSound */ LEGACY_EXEC
 #ifdef _MOBILE
```
No changes in LEGACY_EXEC

runrevmark commented on engine/src/mblandroidmm.cpp:
```diff
@@ -42,29 +42,28 @@
 
-
-	if (MCCStringBeginsWith(p_file, "http://") || MCCStringBeginsWith(p_file, "https://"))
+	if (MCStringBeginsWith(p_video, MCSTR("http://"), kMCStringOptionCompareExact)
```
You can use MCStringBeginsWithCString here

runrevmark commented on engine/src/mblandroidmm.cpp:
```diff
@@ -75,7 +74,7 @@ bool MCSystemPlayVideo(const char *p_file)
 
 	s_video_playing = true;
-	MCAndroidEngineRemoteCall("playVideo", "bsbbbb", &t_success, t_video_path, t_is_asset, t_is_url, t_looping, t_show_controller);
+	MCAndroidEngineRemoteCall("playVideo", "bsbbbb", &t_success, MCStringGetCString(t_video_path), t_is_asset, t_is_url, t_looping, t_show_controller);
```
You can use 'x' as the format specifier for an MCStringRef argument.

runrevmark commented on engine/src/mbliphonesound.mm:
```diff
@@ -98,7 +98,7 @@ - (void)dealloc;
 static MCSoundPlayerDelegate *s_sound_player_delegate = nil;
 static float s_sound_loudness = 1.0;
-static MCStringRef s_sound_file = nil;
+static MCStringRef s_sound_file = MCValueRetain(kMCEmptyString);
```
This initialization needs to be moved to an MCSystemSoundInitialize function with a corresponding MCSystemSoundFinalize function. (Avoid non-constant initializers in statics and globals).

runrevmark commented on engine/src/mbliphonesound.mm:
```diff
@@ -309,16 +296,14 @@ bool MCSystemPlaySound(const char *p_file, bool p_looping)
     {
         // Check if we are playing an ipod file or a resource file.
-        if (strncmp(p_file, "ipod-library://", 15) == 0)
+		if (MCStringBeginsWith(p_sound, MCSTR("ipod-library://"), kMCStringOptionCompareExact))
```
You can use MCStringBeginsWithCString here.

runrevmark commented on engine/src/mbliphonevideo.mm:
```diff
@@ -293,20 +293,28 @@ static bool play_fullscreen_movie_new(NSURL *p_movie, bool p_looping, bool p_wit
 	t_url = nil;
-	if (strncmp(p_file, "http://", 7) == 0 || strncmp(p_file, "https://", 8) == 0)
-		t_url = [NSURL URLWithString: [NSString stringWithCString: p_file encoding: NSMacOSRomanStringEncoding]];
+	if (MCStringBeginsWith(p_video, MCSTR("http://"), kMCStringOptionCompareExact)
```
You can use MCStringBeginsWithCString here.

Commit:
Revert edits to cmdse.cpp and funcs.cpp

Accidental edits made to code within an #ifdef LEGACY_EXEC block.

Commit:
Use MCStringBeginsWithCString where appropriate

Replaced calls to MCStringBeginsWith using a C string with calls to the more appropriate function.

Commit:
Fix an incorrect function signature

The extern declaration for MCSystemGetPlayingSound in MCMultimediaEvalSound had the wrong signature.

Commit:
Add MCSystemSound{Initialize,Finalize} functions

Both Android and iOS now call these functions at startup/shutdown.
